src/assets/wise5/authoringTool/edit-branch/edit-branch.component.ts

```diff
@@ -0,0 +1,216 @@
+  }
+
+  private populateChoiceBranchParam(branch: any, criterion: any): void {
+    const params = criterion.params;
+    if (params.choiceIds.length > 0) {
+      branch.choiceId = params.choiceIds[0];
+    }
+    if (params.nodeId && params.componentId) {
+      const choices = this.projectService.getChoices(params.nodeId, params.componentId);
+      branch.choices = copy(choices);
+    }
+  }
+
+  private getBranchCriteria(): string {
+    const criteria = this.getCriteriaFromTransitions();
+    return criteria == null ? this.node.transitionLogic.howToChooseAmongAvailablePaths : criteria;
+  }
+
+  private getCriteriaFromTransitions(): string {
+    for (const transition of this.node.transitionLogic.transitions) {
+      const criteria = transition.criteria;
+      if (criteria != null) {
+        for (const criterion of transition.criteria) {
+          if (criterion.name === this.SCORE_VALUE) {
+            return this.SCORE_VALUE;
+          } else if (criterion.name === this.CHOICE_CHOSEN_VALUE) {
+            return this.CHOICE_CHOSEN_VALUE;
+          }
+        }
+      }
+    }
+    return null;
+  }
+
+  private getBranchParamByField(fieldName: string): string {
+    for (const transition of this.node.transitionLogic.transitions) {
+      const criteria = transition.criteria;
+      if (criteria != null) {
+        for (const criterion of transition.criteria) {
+          if (criterion.params != null) {
+            return criterion.params[fieldName];
+          }
+        }
+      }
+    }
+    return null;
+  }
+
+  private updateParamFormControls(criteria: string): void {
+    if (criteria === this.SCORE_VALUE) {
+      this.updateScoreFormControls();
+    } else if (criteria === this.CHOICE_CHOSEN_VALUE) {
+      this.updateChoiceFormControls();
+    }
+  }
+
+  private updateScoreFormControls(): void {
+    // use set timeout to give the form time to populate the path form group controls
+    setTimeout(() => {
+      const scores = this.branchPaths.map((branchPath) => branchPath.scores);
+      if (scores.length > 0) {
+        for (let i = 0; i < scores.length; i++) {
+          this.pathFormGroup.controls[`path-${i + 1}`].setValue(scores[i].join(', '));
+        }
+      }
+    });
+  }
+
+  private updateChoiceFormControls(): void {
+    // use set timeout to give the form time to populate the path form group controls
+    setTimeout(() => {
+      const choices = this.branchPaths.map((branchPath) => branchPath.choiceId);
+      if (choices.length > 0) {
+        for (let i = 0; i < choices.length; i++) {
+          this.pathFormGroup.controls[`path-${i + 1}`].setValue(choices[i]);
+        }
+      }
+    });
+  }
+
+  protected removeBranchButtonClicked(): void {
+    if (
+      confirm(
+        $localize`Are you sure you want to remove this branch?\n\nThe branch structure will be removed but the steps will not be deleted.`
+      )
+    ) {
+      this.deleteBranchService.removeBranch(this.branchPaths, this.targetId);
```

Comment:
Keep function name consistent with deleteBranchService?
```suggestion
      this.deleteBranchService.deleteBranch(this.branchPaths, this.targetId);
```